src/posit/connect/resources.py

```diff
@@ -203,24 +191,23 @@ def find(self, uid) -> T:
         Returns
         -------
         T
-
-        Raises
-        ------
-        ValueError
-            If no record is found.
         """
-        # todo - add some more comments about this
         if self._cache:
+            # Check if the record already exists in the cache.
+            # It is assumed that local cache scan is faster than an additional HTTP request.
             conditions = {self._uid: uid}
             result = self.find_by(**conditions)
-        else:
-            endpoint = self._endpoint + uid
-            response = self._ctx.session.get(endpoint)
-            result = response.json()
-            result = self._create_instance(**result)
-
-        if not result:
-            raise ValueError(f"Failed to find instance where {self._uid} is '{uid}'")
+            if result:
+                return result
+
+        endpoint = self._ctx.url + self._path + uid
+        response = self._ctx.session.get(endpoint)
+        result = response.json()
+        result = self._create_instance(self._path, uid, **result)
+
+        # Invalidate the cache.
+        # It is assumed that the cache is stale since a record exists on the server and not in the cache.
+        self._cache = None
```

Comment:
Why not set directly to `result` as its already an result of `._create_instance()`?

Reply:
`_cache` holds the collection, while `result` is a single instance in this case.